org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/fields/AbstractDetailsPart.java

```diff
@@ -104,8 +104,11 @@ public final void propertyChange(PropertyChangeEvent event) {
 		if (event.getSource() instanceof FieldEditor) {
 			FieldEditor fe = (FieldEditor) event.getSource();
 			internalStore.setValue(fe.getPreferenceName(), fe.getPreferenceStore().getDefaultString(fe.getPreferenceName()));
+			// store changed value in internalStore
 			fe.setPreferenceStore(internalStore);
 			fe.store();
+			// reset preference store of fieldEditor to chained store, so it also works for other token styles.
+			fe.setPreferenceStore(getPreferenceStore());
```

Comment:
Shouldn't this be done in a finally block? I assume store can throw either Core or IOExceptions

Reply:
Maybe, but this is not done for similar code in this class.

I tried to keep the fix as small as possible